lib/adapter.js

```diff
@@ -3652,43 +3655,39 @@ function Adapter(options) {
                 if (err) {
                     return typeof callback === 'function' && callback(err);
                 }
+
                 if (res && res.rows) {
-                    let count = 0;
+                    const promises = [];
                     for (let i = 0; i < res.rows.length; i++) {
-                        count++;
-                        adapterObjects.getObject(res.rows[i].id, options, (err, obj) => {
-                            if (err) {
-                                typeof callback === 'function' && callback(err);
-                                callback = null;
-                                return;
-                            } else if (obj && obj.common && obj.common.members) {
-                                const pos = obj.common.members.indexOf(objId);
-                                if (pos !== -1) {
-                                    obj.common.members.splice(pos, 1);
-                                    count++;
-                                    obj.from = 'system.adapter.' + this.namespace;
-                                    obj.user = (options ? options.user : '') || 'system.user.admin';
-                                    obj.ts = Date.now();
-
-                                    adapterObjects.setObject(obj._id, obj, options, err => {
-                                        if (!--count && callback) {
-                                            callback(err);
-                                        } else {
-                                            if (err) {
-                                                typeof callback === 'function' && callback(err);
-                                                callback = null;
-                                            }
-                                        }
-                                    });
+                        promises.push(new Promise(resolve => {
+                            adapterObjects.getObject(res.rows[i].id, options, async (err, obj) => {
+                                if (err) {
+                                    return typeof callback === 'function' && callback(err);
```

Comment:
callback really?

Reply:
That doesn't look correct.

Reply:
As above, existing behaviour. If an error occured the cb was called immediately, if not it was waited until count is zero (all promises fullfilled)

Reply:
In that case we should use reject. Promise.all short-circuits then and we could call the callback in the .catch call.